### PR TITLE
AJ-1602: list-jobs API no longer a passthrough to Import Service

### PR DESCRIPTION
**Pull request by davidangb**
AJ-1602

The eventual goal is to query both cWDS and Import Service, ask each about their running jobs, and aggregate the results into a single response.

This PR is prep for that. Previously, the call to Import Service was a pure passthrough; Orch was blind to Import Service's results and just passed on the response. Now, in this PR, Orch can deserialize the Import Service results and is fully aware of them. We still pass on the results unchanged, but this is a necessary step for when we'll later need to aggregate results.

----

Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- [x] I've followed [the instructions](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [x] I've updated the RC_XXX release ticket with any manual steps required to release this change
- [x] I've updated the [FISMA documentation](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/CONTRIBUTING.md#fisma-documentation-changes) if I've made any security-related changes, including auth, encryption, or auditing

In all cases:

- [ ] Get two thumbsworth of review and PO signoff if necessary
- [ ] Verify all tests go green
- [ ] Squash and merge; you can delete your branch after this **unless it's for a hotfix**. In that case, don't delete it!
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockImportServiceDAO.scala [hunk 1]
```diff
@@ -18,10 +22,13 @@ class MockImportServiceDAO extends ImportServiceDAO {
...
-        else if(importRequest.url.contains("bad.request")) Future.successful(RequestComplete(BadRequest, "Bad request as reported by import service"))
-        else if(importRequest.url.contains("its.lawsuit.time")) Future.successful(RequestComplete(UnavailableForLegalReasons, "import service message"))
-        else if(importRequest.url.contains("good")) Future.successful(RequestComplete(Accepted,
+        if (importRequest.url.contains("forbidden")) Future.successful(RequestComplete(Forbidden, "Missing " +
+          "Authorization: Bearer token in header"))
+        else if (importRequest.url.contains("bad.request")) Future.successful(RequestComplete(BadRequest, "Bad " +
+          "request as reported by import service"))
+        else if (importRequest.url.contains("its.lawsuit.time")) Future.successful(RequestComplete
+        (UnavailableForLegalReasons, "import service message"))
+        else if (importRequest.url.contains("good")) Future.successful(RequestComplete(Accepted,
```
Comment (davidangb) [inline src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockImportServiceDAO.scala:31]: no changes here, just IntelliJ auto-reformatting

File: src/test/scala/org/broadinstitute/dsde/firecloud/webservice/WorkspaceApiServiceJobSpec.scala [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+              verify(mockitoImportServiceDAO, times(1)).listJobs(ArgumentMatchers.eq(workspace.namespace),
+                ArgumentMatchers.eq(workspace.name), ArgumentMatchers.eq(runningOnly))(any[UserInfo])
+            }
+          }
+        }
+      }
+    }
+  }
+
+}
```
Comment (davidangb) [inline src/test/scala/org/broadinstitute/dsde/firecloud/webservice/WorkspaceApiServiceJobSpec.scala:80]: this replaces the unit tests from WorkspaceApiServiceSpec, just below

File: src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala [hunk 3]
```diff
@@ -373,6 +373,14 @@ class EntityService(rawlsDAO: RawlsDAO, importServiceDAO: ImportServiceDAO, goog
...
   }
 
+  def listJobs(workspaceNamespace: String, workspaceName: String, runningOnly: Boolean, userInfo: UserInfo): Future[List[ImportServiceListResponse]] = {
+    // get jobs from Import Service
+    val importServiceJobs = importServiceDAO.listJobs(workspaceNamespace, workspaceName, runningOnly)(userInfo)
+    // TODO AJ-1602: get jobs from cWDS
+    // TODO AJ-1602: merge lists before replying
+    importServiceJobs
+  }
+
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/EntityService.scala:383]: This is where, in future PRs, we'll do all the work of aggregating cWDS and Import Service job info

File: src/main/scala/org/broadinstitute/dsde/firecloud/webservice/WorkspaceApiService.scala [hunk 4]
```diff
@@ -174,9 +177,13 @@ trait WorkspaceApiService extends FireCloudRequestBuilding with FireCloudDirecti
...
-                    requireUserInfo() { _ =>
-                      extract(_.request.uri.query()) { query =>
-                        passthrough(Uri(encodeUri(s"${FireCloudConfig.ImportService.server}/$workspaceNamespace/$workspaceName/imports")).withQuery(query), HttpMethods.GET)
+                    requireUserInfo() { userInfo =>
+                      parameter(Symbol("running_only").as[Boolean].withDefault(false)) { runningOnly =>
+                        complete {
+                          entityServiceConstructor(FlexibleModelSchema).listJobs(workspaceNamespace, workspaceName, runningOnly, userInfo) map { respBody =>
+                            RequestComplete(OK, respBody)
+                          }
+                        }
```
Comment (davidangb) [inline src/main/scala/org/broadinstitute/dsde/firecloud/webservice/WorkspaceApiService.scala:186]: this replaces the previous passthrough (i.e. proxy) implementation for listing jobs. Now, we make the REST call, deserialize the results, then reply with those results. This is prep for future PRs, in which we'll need to interpret the results in order to aggregate cWDS and Import Service values.

**Review by jladieu — APPROVED** [review]
Cool!

**Review by nawatts — APPROVED** [review]

**Review by ashanhol — APPROVED** [review]
changes seem to make sense, can't review on the scala itself

**Pull request merged**